src/safeds/exceptions/_generic.py

```diff
@@ -0,0 +1,151 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+
+
+class OutOfBoundsError(ValueError):
+    """A generic exception that can be used to signal that a (float) value is outside its expected range."""
+
+    def __init__(self, actual: float, *, lower_bound: Bound | None = None, upper_bound: Bound | None = None):
+        """
+        Initialize an OutOfBoundError.
+
+        Parameters
+        ----------
+        actual: float
+            The actual value that is outside its expected range.
+        lower_bound: Bound | None
+            The lower bound of the expected range.
+        upper_bound: Bound | None
+            The upper bound of the expected range.
+        """
+        if (lower_bound is None or isinstance(lower_bound, MinInfinity)) and (
+            upper_bound is None or isinstance(upper_bound, Infinity)
+        ):
+            raise NotImplementedError("Illegal interval: Value cannot be out of bounds if there are no bounds.")
```

Comment:
`NotImplementedError` should not be used here:

> In user defined base classes, abstract methods should raise this exception when they require derived classes to override the method, or while the class is being developed to indicate that the real implementation still needs to be added.

https://docs.python.org/3/library/exceptions.html#NotImplementedError

Either raise a `ValueError` or use the [`assert` statement](https://docs.python.org/3/reference/simple_stmts.html#the-assert-statement).

Reply:
Changed to `ValueError`.

Thinking more about it, I guess one could even just raise the `OutOfBoundsError` normally, but additionally raise a warning?

Reply:
> Thinking more about it, I guess one could even just raise the `OutOfBoundsError` normally, but additionally raise a warning?

I prefer the value error, becasue it might be hard to link the warning and the out of bounds error together for users.

Reply:
Ok, leaving it as `ValueError` then.

I did now, however, include the bounds that were actually given in the error message because the error otherwise seems pretty useless / hard-to-debug.